README.md

```diff
@@ -30,8 +30,13 @@ Second exercise was to add pre-commit hooks and make sure to run them
         -   git commit -m "commit message"
 
     Push to origin:
-        -   git push -u origin {branch-name}
+        -   git push (-u) origin {branch-name}  # only use -u if branch is on GitHub
 
     Create Pull Request on GitHub webapp:
-        -   Under "Pull requests", a new notification should show up to "Compare & pull requests". Click this.
-        -   In order to merge without PR, use `git merge main`. YOU DO NOT WANT TO DO THIS!!! It merges your personal remote branch to the main without review.
\ No newline at end of file
+        -   Under "Pull requests", a new notification should show up to "Compare & pull requests". Click this.  # refer to PR#406
+        -   In order to merge without PR, use `git merge main`. YOU DO NOT WANT TO DO THIS!!! It merges your personal remote branch to the main without review.
+
```

Comment:
Please add the section `How to setup local environment`
and link  https://pre-commit.com/#install in the step of `- install pre-commit hook`